cpu/cpu.go

```diff
@@ -218,43 +225,25 @@ func (p *Plugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.Metri
 			if err != nil {
 				return metrics, err
 			}
-			metric := plugin.MetricType{
-				Namespace_: ns,
-				Data_:      val,
-				Timestamp_: ts,
-				Version_:   version,
+			metric := plugin.Metric{
+				Namespace: ns,
+				Data:      val,
+				Timestamp: ts,
+				Version:   Version,
 			}
 			metrics = append(metrics, metric)
 		}
 	}
 	return metrics, nil
 }
 
-// GetConfigPolicy returns config policy
-// It returns error in case retrieval was not successful
-func (p *Plugin) GetConfigPolicy() (*cpolicy.ConfigPolicy, error) {
-	cp := cpolicy.New()
-	rule, _ := cpolicy.NewStringRule("proc_path", false, "/proc")
-	node := cpolicy.NewPolicyNode()
-	node.Add(rule)
-	cp.Add([]string{vendor, fs, pluginName}, node)
-	return cp, nil
-}
-
-//Meta returns meta data for plugin
-func Meta() *plugin.PluginMeta {
-	return plugin.NewPluginMeta(
-		pluginName,
-		version,
-		pluginType,
-		[]string{},
-		[]string{plugin.SnapGOBContentType},
-		plugin.ConcurrencyCount(1))
-}
-
-func (p *Plugin) init(cfg map[string]ctypes.ConfigValue) error {
-	if procPath, ok := cfg["proc_path"]; ok {
-		p.proc_path = procPath.(ctypes.ConfigValueStr).Value + "/stat"
+func (p *CPUCollector) init(cfg plugin.Config) error {
+	if _, ok := cfg["proc_path"]; ok {
```

Comment:
Like I said before, this is not needed. `cfg.GetString()` is all we need. Regarding tests failing without this - tests are calling `CollectMetrics()` and supplying metrics with no config as an argument. Mocked metrics in tests will need `proc_path` config value set to default value. Alternatively, we could fallback to default value of `p.proc_path` if there is an error in `cfg.GetString()` call, but this approach would be test-specific, as metrics supplied by Snap should always carry default `proc_path` value and thus should be avoided.